tests/test_raft.rs

```diff
@@ -1525,6 +1525,152 @@ fn test_non_promotable_voter_wich_check_quorum() {
 }
 
 #[test]
+fn test_read_only_option_safe() {
+    let a = new_test_raft(1, vec![1, 2, 3], 10, 1, new_storage());
+    let b = new_test_raft(2, vec![1, 2, 3], 10, 1, new_storage());
+    let c = new_test_raft(3, vec![1, 2, 3], 10, 1, new_storage());
+
+    let mut nt = Network::new(vec![Some(a), Some(b), Some(c)]);
+
+    // we can not let system choose the value of randomizedElectionTimeout
+    // otherwise it will introduce some uncertainty into this test case
+    // we need to ensure randomizedElectionTimeout > electionTimeout here
+    let b_election_timeout = nt.peers.get(&2).unwrap().get_election_timeout();
+    nt.peers.get_mut(&2).unwrap().set_randomized_election_timeout(b_election_timeout + 1);
+
+    for _ in 0..b_election_timeout {
+        nt.peers.get_mut(&2).unwrap().tick();
+    }
+    nt.send(vec![new_message(1, 1, MessageType::MsgHup, 0)]);
+
+    assert_eq!(nt.peers.get(&1).unwrap().state, StateRole::Leader);
+
+    let mut tests = vec![
+        (2, 10, 11, "ctx1"),
+        (3, 10, 21, "ctx2"),
+        (2, 10, 31, "ctx3"),
+        (3, 10, 41, "ctx4"),
+    ];
+
+    for (i, (id, proposals, wri, wctx)) in tests.drain(..).enumerate() {
+        for _ in 0..proposals {
+            nt.send(vec![new_message(1, 1, MessageType::MsgPropose, 1)]);
```

Comment:
?? What empty entry exactly?
`new_message` here returns a propose message of 1 entry, with data `SOME_DATA`.